pkg/types/conversion/installconfig_test.go

```diff
@@ -393,6 +393,39 @@ func TestConvertInstallConfig(t *testing.T) {
 			},
 			expectedError: "cannot specify computeFlavor and type in defaultMachinePlatform together",
 		},
+		// NOTE: THIS DOES NOT WORK YET
+		// {
+		// 	name: "deprecated OpenStack controlPlane with type in rootVolume",
+		// 	config: &types.InstallConfig{
+		// 		TypeMeta: metav1.TypeMeta{
+		// 			APIVersion: types.InstallConfigVersion,
+		// 		},
+		// 		ControlPlane: &types.MachinePool{
+		// 			Platform: types.MachinePoolPlatform{
+		// 				OpenStack: &openstack.MachinePool{
+		// 					RootVolume: &openstack.RootVolume{
+		// 						DeprecatedType: "fast",
+		// 					},
+		// 				},
+		// 			},
+		// 		},
+		// 	},
+		// 	expected: &types.InstallConfig{
+		// 		TypeMeta: metav1.TypeMeta{
+		// 			APIVersion: types.InstallConfigVersion,
+		// 		},
+		// 		ControlPlane: &types.MachinePool{
+		// 			Platform: types.MachinePoolPlatform{
+		// 				OpenStack: &openstack.MachinePool{
+		// 					RootVolume: &openstack.RootVolume{
+		// 						DeprecatedType: "",
+		// 						Types:          []string{"fast"},
+		// 					},
+		// 				},
+		// 			},
+		// 		},
+		// 	},
+		// },
```

Comment:
That's weird, but conversion is only triggered if there is an OpenStack platform. I have added one.

Also, you are not zeroing the deprecated type, so we should expect it back.

```diff
diff --git a/pkg/types/conversion/installconfig_test.go b/pkg/types/conversion/installconfig_test.go
index cdc1cb4ed2..e09b929f93 100644
--- a/pkg/types/conversion/installconfig_test.go
+++ b/pkg/types/conversion/installconfig_test.go
@@ -399,6 +399,7 @@ func TestConvertInstallConfig(t *testing.T) {
 				TypeMeta: metav1.TypeMeta{
 					APIVersion: types.InstallConfigVersion,
 				},
+				Platform: types.Platform{OpenStack: &openstack.Platform{}},
 				ControlPlane: &types.MachinePool{
 					Platform: types.MachinePoolPlatform{
 						OpenStack: &openstack.MachinePool{
@@ -413,11 +414,12 @@ func TestConvertInstallConfig(t *testing.T) {
 				TypeMeta: metav1.TypeMeta{
 					APIVersion: types.InstallConfigVersion,
 				},
+				Platform: types.Platform{OpenStack: &openstack.Platform{}},
 				ControlPlane: &types.MachinePool{
 					Platform: types.MachinePoolPlatform{
 						OpenStack: &openstack.MachinePool{
 							RootVolume: &openstack.RootVolume{
-								DeprecatedType: "",
+								DeprecatedType: "fast",
 								Types:          []string{"fast"},
 							},
 						},
```